### PR TITLE
docs: improve proxy docs (ALL_PROXY vs HTTPS_PROXY)

### LINKED ISSUE #1023: cloud_sql_proxy returns errors parsing config when running behind SOCKS5 proxy?
## Question
cloud_sql_proxy returns `errors parsing config` when running behind SOCKS5 proxy.

## Additional Context

I am running  cloud_sql_proxy behind [ShadowsocksX-NG](https://github.com/shadowsocks/ShadowsocksX-NG/releases) at MacOS 12.0.1. I have set up a local socks5 proxy using shadowsocksX-NG.
Screen Shot 2021-11-23 at 16 04 35

I have tested the socks5 proxy by curl, It seems OK.
```bash
➜  ~ curl -x socks5://127.0.0.1:1086 www.example.com
<!doctype html>
<html>
<head>
    <title>Example Domain</title>

    <meta charset="utf-8" />
    <meta http-equiv="Content-type" content="text/html; charset=utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type="text/css">
    body {
        background-color: #f0f0f2;
        margin: 0;
        padding: 0;
        font-family: -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
        
    }
...
```
Then, I have configured the proxy using `gcloud config`

```ini
➜  ~ gcloud config list           
...
[proxy]
address = 127.0.0.1
port = 1086
type = socks5

Your active configuration is: [default]  
```
the gcloud info --run-diagnostics can pass, so does gcloud init

```bash
-> gcloud info --run-diagnostics
Network diagnostic detects and fixes local network connection issues.
Checking network connection...done.                                                                        
Reachability Check passed.
Network diagnostic passed (1/1 checks passed).

Property diagnostic detects issues that may be caused by properties.
Checking hidden properties...done.                                                                         
Hidden Property Check passed.
Property diagnostic passed (1/1 checks passed).
```

I follow the doc at [cloudsql-proxy](https://github.com/GoogleCloudPlatform/cloudsql-proxy#running-behind-a-socks5-proxy), it still not working.

```bash
-> ALL_PROXY=socks5://127.0.0.1:1086 cloud_sql_proxy -instances="xxx:us-west1:xxx"=tcp:3306
2021/11/23 16:12:05 Rlimits for file descriptors set to {&{8500 9223372036854775807}}
2021/11/23 16:12:37 errors parsing config:
	Get "https://sqladmin.googleapis.com/sql/v1beta4/projects/xxx/instances/xxx?alt=json&prettyPrint=false": dial tcp 172.217.163.42:443: i/o timeout 
```

```json
{
  "error": {
    "code": 401,
    "message": "Request is missing required authentication credential. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project.",
    "errors": [
      {
        "message": "Login Required.",
        "domain": "global",
        "reason": "required",
        "location": "Authorization",
        "locationType": "header"
      }
    ],
    "status": "UNAUTHENTICATED",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        "reason": "CREDENTIALS_MISSING",
        "domain": "googleapis.com",
        "metadata": {
          "method": "google.cloud.sql.v1beta4.SqlInstancesService.Get",
          "service": "sqladmin.googleapis.com"
        }
      }
    ]
  }
}
```

The `xxx` represents the instance/project name. 
My colleague has tried `cloud_sql_proxy -instances="xxx:us-west1:xxx"=tcp:3306` and it works fine.
Also, I have used `cloud_sql_proxy -instances="xxx:us-west1:xxx"=tcp:3306` before and it worked fine too.

Do you have an idea of the problem or how to gather more information to debug the issue?

### PR DESCRIPTION
**Pull request by enocom**
Fixes #1023.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -304,9 +304,17 @@ proxy. If a Socks5 proxy is running on `localhost:8000`, the command to start
...
 
 ```
-ALL_PROXY=socks5://localhost:8000 cloud_sql_proxy -instances=$INSTANCE_CONNECTION_NAME=tcp:5432
+ALL_PROXY=socks5://localhost:8000 \
+HTTPS_PROXY=socks5://localhost:8000 \
+    cloud_sql_proxy -instances=$INSTANCE_CONNECTION_NAME=tcp:5432
 ```
 
+Note: the `ALL_PROXY` environment variable specifies the proxy for all TCP
+traffic to and from a Cloud SQL instance. The `HTTPS_PROXY` (or `HTTP_PROXY`)
```
Comment (kurtisvg) [inline README.md:313]: Does this need to mention `socks5h` as well? That's the mode that lets the socks5 proxy resolve the dns rather than the client

Comment (enocom) [reply README.md]: Fixed.

File: README.md [hunk 2]
```diff
@@ -299,14 +299,25 @@ Kubernetes Engine][connect-to-k8s].
...
+HTTPS_PROXY=socks5://localhost:8000 \
+    cloud_sql_proxy -instances=$INSTANCE_CONNECTION_NAME=tcp:5432
 ```
 
+Note: the `ALL_PROXY` environment variable specifies the proxy for all TCP
+traffic to and from a Cloud SQL instance. The `ALL_PROXY` environment variable
+supports `socks5` and `socks5h` protocols. To route DNS lookups through a proxy,
+use the `socks5h` protocol.
+
+The `HTTPS_PROXY` (or `HTTP_PROXY`) specifies the proxy for all HTTP(S) traffic. See
```
Comment (kurtisvg) [inline README.md:317]: Can we specify that `HTTPS_PROXY` is for interactions the SQLAdmin API? They also do not need to use HTTPS unless they specifically want their API calls to go through the socks5 proxy

**Review by kurtisvg — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (enocom) [reply README.md]: Done.

**Pull request merged**